pkg/sdn/plugin/multitenant.go

```diff
@@ -146,52 +141,20 @@ func (mp *multiTenantPlugin) RefVNID(vnid uint32) {
 	}
 }
 
-func (mp *multiTenantPlugin) UnrefVNID(vnid uint32) {
+func (mp *multiTenantPlugin) RemoveVNIDRules(vnid uint32) {
 	if vnid == 0 {
 		return
 	}
 
-	mp.vnidRefsLock.Lock()
-	defer mp.vnidRefsLock.Unlock()
-	if mp.vnidRefs[vnid] == 0 {
-		glog.Warningf("refcounting error on vnid %d", vnid)
-		return
-	}
-	mp.vnidRefs[vnid] -= 1
-	if mp.vnidRefs[vnid] > 0 {
-		return
-	}
-	glog.V(5).Infof("UnrefVNID %d removing rule", vnid)
+	mp.vnidInUseLock.Lock()
+	defer mp.vnidInUseLock.Unlock()
+	mp.vnidInUse[vnid] = false
+
+	glog.V(5).Infof("RemoveVNIDRules %d", vnid)
 
 	otx := mp.node.oc.NewTransaction()
-	otx.DeleteFlows("table=80, reg0=%d, reg1=%d", vnid, vnid)
+	otx.DeleteFlows("table=80, reg1=%d", vnid)
```

Comment:
Why did you loosen this up?  (I'm not really concerned by the change, I just want to understand the reason).  Will it make it harder to match this up with the AddFlow that we do?  I assume it corresponds with the one on the new line 138.